assets/wire/components/WireApp.jsx

```diff
@@ -48,7 +50,13 @@ class WireApp extends React.Component {
     renderModal(specs) {
         if (specs) {
             const Modal = modals[specs.modal];
-            return <Modal key="modal" data={specs.data} />;
+            return <Modal
+                key="modal"
+                data={specs.data}
+                closeModal={this.props.closeModal}
+                submit={this.props.submitModal}
```

Comment:
imo this part is confusing, any modal would get this while it only works for follow topic modal, thus I think it would be better to make that component smart and let it pick the action, how we avoid the duplication in actions/reducers is another problem

Reply:
I see. Let me think about this component.